sdk/test/trace/tracer_test.cc

```diff
@@ -238,3 +241,125 @@ TEST(Tracer, SpanSetAttribute)
   auto &span_data = spans_received->at(0);
   ASSERT_EQ(3.1, nostd::get<double>(span_data->GetAttributes().at("abc")));
 }
+
+TEST(Tracer, TestAlwaysOnSampler)
+{
+  std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_on(
+      new std::vector<std::unique_ptr<SpanData>>);
+  
+  auto tracer_on = initTracer(spans_received_on);
+
+  // Testing AlwaysOn sampler.
+  // Create two spans for each tracer. Check the exported result.
+  auto span_on_1 = tracer_on->StartSpan("span with AlwaysOn sampler 1");
+  auto span_on_2 = tracer_on->StartSpan("span with AlwaysOn sampler 2");
+
+  span_on_1->SetAttribute("attr1", 3.1);
+
+  span_on_2->End();
+  span_on_1->End();
+  ASSERT_EQ(2, spans_received_on->size());
+  auto &span_data_on_1 = spans_received_on->at(1);  // span 2 ends first.
+  auto &span_data_on_2 = spans_received_on->at(0);
+  ASSERT_EQ(3.1, nostd::get<double>(span_data_on_1->GetAttributes().at("attr1")));
+  ASSERT_LT(std::chrono::nanoseconds(0), span_data_on_1->GetStartTime().time_since_epoch());
+  ASSERT_LT(std::chrono::nanoseconds(0), span_data_on_1->GetDuration());
+  ASSERT_LT(std::chrono::nanoseconds(0), span_data_on_2->GetStartTime().time_since_epoch());
+  ASSERT_LT(std::chrono::nanoseconds(0), span_data_on_2->GetDuration());
+}
+
+TEST(Tracer, TestAlwaysOffSampler)
+{
+  std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_off(
+      new std::vector<std::unique_ptr<SpanData>>);
+  auto tracer_off = initTracer(spans_received_off, std::make_shared<AlwaysOffSampler>());
+  // Testing AlwaysOff sampler
```

Comment:
We can also omit those comments. I think that's clear enough from the test name.